<h1>Содержание<span class="tocSkip"></span></h1>

In [23]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
!pip install xlrd
!pip install openpyxl

In [24]:
df_1 = pd.read_excel('Тестовое_задание_для_кандидатов_2022.xlsx', engine='openpyxl', sheet_name = 'Задание 1', skiprows = 4)
df_1.info()
#Удаляем пустые строки, для того чтобы данные были полные
df_1 = df_1.dropna(subset=['Категория устройства'])
df_1.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14636 entries, 0 to 14635
Data columns (total 6 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   ID абонента                    14636 non-null  int64  
 1   Категория устройства           14589 non-null  object 
 2   Сумма начислений в месяц, руб  14636 non-null  float64
 3   Трафик голосовой, мин          14636 non-null  float64
 4   Трафик передачи данных, Мб     14636 non-null  float64
 5   Трафик в роуминге, мин         14636 non-null  float64
dtypes: float64(4), int64(1), object(1)
memory usage: 629.0+ KB


(14589, 6)

In [25]:
df_1['Категория устройства'] = df_1['Категория устройства'].str.lower()
df_1['Сумма начислений в месяц, руб'] = df_1['Сумма начислений в месяц, руб'].round(1)
df_1['Трафик голосовой, мин'] = df_1['Трафик голосовой, мин'].round(1)
df_1['Трафик передачи данных, Мб'] = df_1['Трафик передачи данных, Мб'].round(1)
df_1['Трафик в роуминге, мин'] = df_1['Трафик в роуминге, мин'].round(1)
df_1['Трафик голосовой, дни'] = (df_1['Трафик голосовой, мин']/(60*24)).round(0)
df_1['Трафик передачи данных, Гб'] = (df_1['Трафик передачи данных, Мб']/1024).round(0)
df_1

ID абонента Категория устройства  Сумма начислений в месяц, руб  \
0          4277642           smartphone                        11962.0   
1          4277933           smartphone                        13864.6   
2          4278138           smartphone                         3905.1   
3          4281994           smartphone                        12475.7   
4          4282050           smartphone                         9519.5   
...            ...                  ...                            ...   
14631    495162131            data card                            2.2   
14632    495164487            data card                            2.2   
14633    495454612            data card                           14.4   
14634    498468449            data card                            0.0   
14635    498469004            data card                            0.0   

       Трафик голосовой, мин  Трафик передачи данных, Мб  \
0                     1003.6                     13975.7   
1                     1159.1                      3619.6   
2                     6300.6                      2298.2   
3                     1845.7                      6296.3   
4                     6565.5                      2576.8   
...                      ...                         ...   
14631                    0.0                         0.0   
14632                    0.0                         0.0   
14633                    0.0                         0.7   
14634                    0.0                         0.0   
14635                    0.0                         0.0   

       Трафик в роуминге, мин  Трафик голосовой, дни  \
0                         2.0                    1.0   
1                        22.8                    1.0   
2                         0.0                    4.0   
3                        70.3                    1.0   
4                         1.7                    5.0   
...                       ...                    ...   
14631                     0.0                    0.0   
14632                     0.0                    0.0   
14633                     0.0                    0.0   
14634                     0.0                    0.0   
14635                     0.0                    0.0   

       Трафик передачи данных, Гб  
0                            14.0  
1                             4.0  
2                             2.0  
3                             6.0  
4                             3.0  
...                           ...  
14631                         0.0  
14632                         0.0  
14633                         0.0  
14634                         0.0  
14635                         0.0  

[14589 rows x 8 columns]

Проверим голосовой трафик на выбросы, будем считать месяц как 30 дней, все что выше уберем из датасета
Принимаем такие условности, так как неуказано может ли с одного ID абонента совершать звонки несколько пользователей
Будем считать что один ID - один пользователь, который физически не может говорить больше 30 дней

In [26]:
df_1 = df_1[df_1['Трафик голосовой, дни'] < 30]
df_1.describe()

ID абонента  Сумма начислений в месяц, руб  Трафик голосовой, мин  \
count  1.457800e+04                   1.457800e+04           14578.000000   
mean   2.631294e+08                   9.928233e+04             942.992317   
std    1.495649e+08                   1.245207e+07            1985.205666   
min    4.277642e+06                  -1.003596e+08            -541.000000   
25%    1.426857e+08                   3.140000e+02               3.900000   
50%    1.513043e+08                   7.527500e+02              58.000000   
75%    4.148311e+08                   1.751075e+03             969.250000   
max    4.984690e+08                   1.500098e+09           37160.000000   

       Трафик передачи данных, Мб  Трафик в роуминге, мин  \
count                1.457800e+04            14578.000000   
mean                 2.361797e+04               93.398429   
std                  4.827905e+04              309.150712   
min                 -5.193740e+04               -1.000000   
25%                  5.317750e+02                0.000000   
50%                  3.080100e+03                0.000000   
75%                  4.510740e+04                0.500000   
max                  1.772188e+06             6782.900000   

       Трафик голосовой, дни  Трафик передачи данных, Гб  
count           14578.000000                14578.000000  
mean                0.644396                   23.056043  
std                 1.398667                   47.158722  
min                 0.000000                  -51.000000  
25%                 0.000000                    1.000000  
50%                 0.000000                    3.000000  
75%                 1.000000                   44.000000  
max                26.000000                 1731.000000

Удалим трафик передачи данных (Гб) который больше 23 гб (принимаю среднее значение ) и сумма начислений равно 0

In [27]:
df_1 = df_1[~((df_1['Трафик передачи данных, Гб'] > 23) & (df_1['Сумма начислений в месяц, руб'] == 0))]

Заменим отрицательные значения в столбцах голосового трафика, трафика передачи данных и трафика в роуминге на медианные значения, так как примем что эти величины не могут быть отрицательные

In [28]:
df_1.loc[df_1["Трафик голосовой, мин"] < 0, "Трафик голосовой, мин"] = \
    df_1.loc[df_1["Трафик голосовой, мин"] >= 0, "Трафик голосовой, мин"].median()
df_1.loc[df_1["Трафик передачи данных, Мб"] < 0, "Трафик передачи данных, Мб"] = \
    df_1.loc[df_1["Трафик передачи данных, Мб"] >= 0, "Трафик передачи данных, Мб"].median()
df_1.loc[df_1["Трафик в роуминге, мин"] < 0, "Трафик в роуминге, мин"] = \
    df_1.loc[df_1["Трафик в роуминге, мин"] >= 0, "Трафик в роуминге, мин"].median()

In [29]:
df_1[df_1['Сумма начислений в месяц, руб'] < 0]

ID абонента Категория устройства  Сумма начислений в месяц, руб  \
2772     136615720            data card                          -44.6   
2777     136640191            data card                          -44.6   
2780     136650318            data card                          -44.6   
2896     137460003            data card                          -33.1   
2898     137467333            data card                          -15.5   
2900     137477458            data card                          -44.6   
2903     137482876            data card                          -43.7   
2904     137484624            data card                          -44.6   
2905     137484946            data card                          -27.5   
2907     137491713            data card                          -44.6   
2908     137491725            data card                          -25.5   
2909     137492657            data card                          -44.6   
3475     141655413            data card                           -0.1   
3508     141775579            data card                           -7.6   
3606     142539417            data card                          -22.4   
3611     142548652            data card                           -0.6   
3622     142564908            data card                          -20.4   
3891     143364652                modem                   -100359563.0   
4451     145194697            data card                           -0.9   
4519     145445864            data card                           -2.3   
4639     145835005            data card                           -0.9   
4883     146390907            data card                           -2.3   
4999     146644476            data card                          -14.1   
6850     150472161            data card                           -2.3   
8588     385815374            data card                           -6.8   
8750     388826777            data card                           -2.3   
9407     395785042            data card                           -7.6   
9410     395787677            data card                           -1.9   
10228    406133667            data card                           -0.1   
11448    423049083            data card                          -36.9   
11678    425228082            data card                          -36.9   
13052    440043968            data card                          -36.9   
13222    442578709            data card                           -0.1   
13232    442600037            data card                          -44.3   
13491    444451753            data card                           -0.2   
13492    444452797            data card                           -0.2   
13494    444466454            data card                           -0.2   
13849    446592922            data card                           -0.6   
13855    446719316            data card                           -0.2   

       Трафик голосовой, мин  Трафик передачи данных, Мб  \
2772                     0.0                         0.0   
2777                     0.0                         0.0   
2780                     0.0                         0.0   
2896                     0.0                         3.3   
2898                     0.0                         8.3   
2900                     0.0                         0.0   
2903                     0.0                         0.2   
2904                     0.0                         0.0   
2905                     0.0                         0.8   
2907                     0.0                         0.1   
2908                     0.0                         5.5   
2909                     0.0                         0.0   
3475                     0.0                         0.0   
3508                     0.0                         0.0   
3606                     0.0                         0.0   
3611                     0.0                         0.0   
3622                     0

In [81]:
df_1[df_1['Сумма начислений в месяц, руб'] > 10**8]

ID абонента Категория устройства  Сумма начислений в месяц, руб  \
10381    409184460                modem                   1.500098e+09   

       Трафик голосовой, мин  Трафик передачи данных, Мб  \
10381                   57.4                     17202.7   

       Трафик в роуминге, мин  Трафик голосовой, дни  \
10381                     0.0                    0.0   

       Трафик передачи данных, Гб  
10381                        17.0

Примем значение Сумма начислений в месяц, руб = -100359563.0 и  1.500098e+09  как аномальное и очистим данные от этого значения

In [87]:
df_1 = df_1[~(df_1['ID абонента'] == 143364652)]
df_1 = df_1[~(df_1['ID абонента'] == 409184460)]
df_1.describe()

ID абонента  Сумма начислений в месяц, руб  Трафик голосовой, мин  \
count  1.457400e+04                   14574.000000           14574.000000   
mean   2.631254e+08                    3266.049787             943.310059   
std    1.495673e+08                   10381.011415            1985.383603   
min    4.277642e+06                     -44.600000               0.000000   
25%    1.426857e+08                     314.125000               3.900000   
50%    1.513043e+08                     752.850000              58.100000   
75%    4.148311e+08                    1751.075000             969.675000   
max    4.984690e+08                  591504.500000           37160.000000   

       Трафик передачи данных, Мб  Трафик в роуминге, мин  \
count                1.457400e+04            14574.000000   
mean                 2.338459e+04               93.424132   
std                  4.371958e+04              309.189243   
min                  0.000000e+00                0.000000   
25%                  5.332000e+02                0.000000   
50%                  3.081300e+03                0.000000   
75%                  4.492245e+04                0.500000   
max                  1.017468e+06             6782.900000   

       Трафик голосовой, дни  Трафик передачи данных, Гб  
count           14574.000000                14574.000000  
mean                0.644573                   22.823521  
std                 1.398818                   42.709825  
min                 0.000000                  -51.000000  
25%                 0.000000                    1.000000  
50%                 0.000000                    3.000000  
75%                 1.000000                   44.000000  
max                26.000000                  994.000000

# Сегментация

In [89]:
(df_1
 .pivot_table(index= 'Категория устройства', 
              values =['Сумма начислений в месяц, руб', 'Трафик голосовой, мин', 'Трафик передачи данных, Мб', 'Трафик в роуминге, мин' ], 
              aggfunc = 'median'
             ).sort_values(by='Сумма начислений в месяц, руб', ascending=False).reset_index()
)

Категория устройства  Сумма начислений в месяц, руб  Трафик в роуминге, мин  \
0                phone                         2804.8                     2.6   
1            usb modem                         2370.7                     0.0   
2           smartphone                         1314.1                     0.0   
3               router                          504.0                     0.0   
4                modem                          489.2                     0.0   
5               tablet                          150.2                     0.0   
6       network device                           71.0                     0.0   
7            data card                            1.7                     0.0   

   Трафик голосовой, мин  Трафик передачи данных, Мб  
0                 6498.5                        0.00  
1                 6402.6                        0.00  
2                  784.6                    12715.10  
3                   21.8                     3060.70  
4                   21.3                     2808.75  
5                 6787.0                      877.90  
6                 7743.6                        0.00  
7                    0.0                        0.00

Сегментация профиля пользователя в зависимости от категории устройства

In [100]:
df_phone = df_1[df_1['Категория устройства'] == 'phone']
df_phone.describe()

ID абонента  Сумма начислений в месяц, руб  Трафик голосовой, мин  \
count  3.270000e+02                     327.000000             327.000000   
mean   2.385535e+08                    9157.829969            6197.869725   
std    1.551794e+08                   34181.921001            3331.162405   
min    4.287114e+06                       0.000000             718.400000   
25%    1.385921e+08                    1090.100000            3222.200000   
50%    1.504465e+08                    2804.800000            6498.500000   
75%    4.055241e+08                   12506.300000            7839.300000   
max    4.558612e+08                  591504.500000           26836.900000   

       Трафик передачи данных, Мб  Трафик в роуминге, мин  \
count                  327.000000               327.00000   
mean                   746.875229               313.95107   
std                   5077.032064               795.44930   
min                      0.000000                 0.00000   
25%                      0.000000                 0.00000   
50%                      0.000000                 2.60000   
75%                      1.150000               172.80000   
max                  84019.300000              5373.50000   

       Трафик голосовой, дни  Трафик передачи данных, Гб  
count             327.000000                  327.000000  
mean                4.308869                    0.718654  
std                 2.307499                    4.966179  
min                 0.000000                    0.000000  
25%                 2.000000                    0.000000  
50%                 5.000000                    0.000000  
75%                 5.000000                    0.000000  
max                19.000000                   82.000000

<div style="border-radius: 15px; border: 3px solid PaleGreen; padding: 15px;">
<b> Сегмент 1</b>
  
    Пользователи, которые пользуются категорией устройства 'phone' чаще всего совершают звонки в сети и в роуминге, реже пользуются интернетом.

</div>

In [122]:
df_smartphone = df_1[df_1['Категория устройства'] == 'smartphone']
df_smartphone.describe()

ID абонента  Сумма начислений в месяц, руб  Трафик голосовой, мин  \
count  8.169000e+03                    8169.000000            8169.000000   
mean   2.440047e+08                    5144.330444            1381.996022   
std    1.528739e+08                   11302.545745            2088.758627   
min    4.277642e+06                       0.000000               0.000000   
25%    1.397592e+08                     780.200000             238.900000   
50%    1.503053e+08                    1314.100000             784.600000   
75%    4.102879e+08                    3268.600000            1409.500000   
max    4.899019e+08                  434140.600000           37160.000000   

       Трафик передачи данных, Мб  Трафик в роуминге, мин  \
count                8.169000e+03             8169.000000   
mean                 3.758995e+04              153.120541   
std                  4.944666e+04              362.289796   
min                  0.000000e+00                0.000000   
25%                  2.020800e+03                0.000000   
50%                  1.271510e+04                0.000000   
75%                  6.281980e+04               25.800000   
max                  1.017468e+06             6782.900000   

       Трафик голосовой, дни  Трафик передачи данных, Гб  
count            8169.000000                 8169.000000  
mean                0.948709                   36.704248  
std                 1.485298                   48.298583  
min                 0.000000                   -4.000000  
25%                 0.000000                    2.000000  
50%                 1.000000                   12.000000  
75%                 1.000000                   61.000000  
max                26.000000                  994.000000

<div style="border-radius: 15px; border: 3px solid PaleGreen; padding: 15px;">
<b> Сегмент 2</b>
  
    Портрет пользователя "smartphone" разнообразный, здесь вовлечены все основные показатели.
    Средние показатели составляют:
        + Трафик передачи данных, Гб - 36.7
        + Трафик голосовой, мин - 1382
        + Трафик в роуминге, мин - 153.1

</div>

In [123]:
df_usb_modem = df_1[df_1['Категория устройства'] == 'usb modem']
df_usb_modem

ID абонента Категория устройства  Сумма начислений в месяц, руб  \
5413    147839828            usb modem                         2370.7   

      Трафик голосовой, мин  Трафик передачи данных, Мб  \
5413                 6402.6                         0.0   

      Трафик в роуминге, мин  Трафик голосовой, дни  \
5413                     0.0                    4.0   

      Трафик передачи данных, Гб  
5413                         0.0

<div style="border-radius: 15px; border: 3px solid PaleGreen; padding: 15px;">
<b> Сегмент 3</b>
  
    "Usb modem" в выборке не пользуется популярность, всего 1 абонент использует для звонков.
    При этом голосовой трафик достигает 6000 мин, что превышает среднее значение более чем в 4 раза 

</div>

In [124]:
df_router = df_1[df_1['Категория устройства'] == 'router']
df_router.describe()

ID абонента  Сумма начислений в месяц, руб  Трафик голосовой, мин  \
count  5.730000e+02                      573.00000             573.000000   
mean   2.845492e+08                      974.24014              53.534555   
std    1.403589e+08                     9554.60168             458.767619   
min    6.427315e+06                        0.00000               0.000000   
25%    1.447802e+08                      325.40000               4.700000   
50%    2.999184e+08                      504.00000              21.800000   
75%    4.239515e+08                      715.80000              40.600000   
max    4.555122e+08                   229069.10000            8759.800000   

       Трафик передачи данных, Мб  Трафик в роуминге, мин  \
count                  573.000000              573.000000   
mean                  8635.629145                3.040314   
std                  19982.991461               66.968914   
min                      0.000000                0.000000   
25%                   1730.000000                0.000000   
50%                   3060.700000                0.000000   
75%                   6771.200000                0.000000   
max                 271272.800000             1602.100000   

       Трафик голосовой, дни  Трафик передачи данных, Гб  
count             573.000000                  573.000000  
mean                0.019197                    8.286213  
std                 0.303790                   19.679116  
min                 0.000000                  -51.000000  
25%                 0.000000                    2.000000  
50%                 0.000000                    3.000000  
75%                 0.000000                    7.000000  
max                 6.000000                  265.000000

<div style="border-radius: 15px; border: 3px solid PaleGreen; padding: 15px;">
<b> Сегмент 4</b>
  
    Пользователи "Router" преимущественно пользуются интернетом
</div>

In [125]:
df_modem = df_1[df_1['Категория устройства'] == 'modem']
df_modem.describe()

ID абонента  Сумма начислений в месяц, руб  Трафик голосовой, мин  \
count  3.040000e+03                    3040.000000            3040.000000   
mean   2.899845e+08                     564.648289              23.602467   
std    1.392115e+08                     458.106419              19.420434   
min    4.454558e+06                       0.000000               0.000000   
25%    1.444543e+08                     315.000000               4.100000   
50%    3.804785e+08                     489.200000              21.300000   
75%    4.233070e+08                     678.025000              41.125000   
max    4.559731e+08                    9179.800000              62.000000   

       Трафик передачи данных, Мб  Трафик в роуминге, мин  \
count                3.040000e+03             3040.000000   
mean                 9.220494e+03                0.234046   
std                  3.473312e+04                1.746928   
min                  1.007900e+03                0.000000   
25%                  1.618200e+03                0.000000   
50%                  2.808750e+03                0.000000   
75%                  6.399550e+03                0.000000   
max                  1.017468e+06               48.800000   

       Трафик голосовой, дни  Трафик передачи данных, Гб  
count                 3040.0                 3040.000000  
mean                     0.0                    8.988158  
std                      0.0                   33.933308  
min                      0.0                    1.000000  
25%                      0.0                    2.000000  
50%                      0.0                    3.000000  
75%                      0.0                    6.000000  
max                      0.0                  994.000000

<div style="border-radius: 15px; border: 3px solid PaleGreen; padding: 15px;">
<b> Сегмент 5</b>
  
    Категория устройства "modem" пользуется популярностью, более 3000 абонентов. 
    При этом абоненты данного вида устройства используют преимущественно трафик передачи данных
</div>

In [128]:
df_tablet = df_1[df_1['Категория устройства'] == 'tablet']
df_tablet.describe()

ID абонента  Сумма начислений в месяц, руб  Трафик голосовой, мин  \
count  4.000000e+00                       4.000000               4.000000   
mean   2.773733e+08                    1551.550000            6784.925000   
std    1.600377e+08                    2852.065937             583.537913   
min    1.305230e+08                      76.500000            6130.500000   
25%    1.434667e+08                     121.725000            6403.950000   
50%    2.759119e+08                     150.200000            6787.000000   
75%    4.098185e+08                    1580.025000            7167.975000   
max    4.271465e+08                    5829.300000            7435.200000   

       Трафик передачи данных, Мб  Трафик в роуминге, мин  \
count                    4.000000                    4.00   
mean                  1080.850000                  794.45   
std                   1291.130279                 1588.90   
min                      0.000000                    0.00   
25%                      0.300000                    0.00   
50%                    877.900000                    0.00   
75%                   1958.450000                  794.45   
max                   2567.600000                 3177.80   

       Трафик голосовой, дни  Трафик передачи данных, Гб  
count                   4.00                        4.00  
mean                    4.75                        1.25  
std                     0.50                        1.50  
min                     4.00                        0.00  
25%                     4.75                        0.00  
50%                     5.00                        1.00  
75%                     5.00                        2.25  
max                     5.00                        3.00

<div style="border-radius: 15px; border: 3px solid PaleGreen; padding: 15px;">
<b> Сегмент 5</b>
  
    Категорией устройства "tablet" пользуется всего 4 человека из выборки. 
    При этом абоненты данного вида устройства активно используют голосовой трафик и трафик в роуминге
</div>

In [131]:
df_network_device = df_1[df_1['Категория устройства'] == 'network device']
df_network_device.describe()

ID абонента  Сумма начислений в месяц, руб  Трафик голосовой, мин  \
count  3.000000e+00                       3.000000               3.000000   
mean   3.099677e+08                      87.733333           13142.666667   
std    1.743251e+08                      79.236439           10445.894967   
min    1.421763e+08                      18.200000            6501.200000   
25%    2.198696e+08                      44.600000            7122.400000   
50%    2.975629e+08                      71.000000            7743.600000   
75%    3.938634e+08                     122.500000           16463.400000   
max    4.901639e+08                     174.000000           25183.200000   

       Трафик передачи данных, Мб  Трафик в роуминге, мин  \
count                         3.0                3.000000   
mean                          0.0                0.233333   
std                           0.0                0.404145   
min                           0.0                0.000000   
25%                           0.0                0.000000   
50%                           0.0                0.000000   
75%                           0.0                0.350000   
max                           0.0                0.700000   

       Трафик голосовой, дни  Трафик передачи данных, Гб  
count               3.000000                         3.0  
mean                9.000000                         0.0  
std                 6.928203                         0.0  
min                 5.000000                         0.0  
25%                 5.000000                         0.0  
50%                 5.000000                         0.0  
75%                11.000000                         0.0  
max                17.000000                         0.0

<div style="border-radius: 15px; border: 3px solid PaleGreen; padding: 15px;">
<b> Сегмент 6</b>
  
    Категорией устройства "network_device" пользуется всего 3 человека из выборки. 
    При этом абоненты данного вида устройства используют голосовой трафик
</div>

In [132]:
df_data_card = df_1[df_1['Категория устройства'] == 'data card']
df_data_card.describe()

ID абонента  Сумма начислений в месяц, руб  Трафик голосовой, мин  \
count  2.457000e+03                    2457.000000            2457.000000   
mean   2.917060e+08                     120.941514             104.263248   
std    1.416717e+08                    1199.110478             874.977248   
min    4.462038e+06                     -44.600000               0.000000   
25%    1.433557e+08                       0.100000               0.000000   
50%    3.788341e+08                       1.700000               0.000000   
75%    4.240549e+08                      18.000000               0.000000   
max    4.984690e+08                   34810.000000           10197.500000   

       Трафик передачи данных, Мб  Трафик в роуминге, мин  \
count                 2457.000000             2457.000000   
mean                   206.633822                0.987993   
std                   6588.495550               47.563352   
min                      0.000000                0.000000   
25%                      0.000000                0.000000   
50%                      0.000000                0.000000   
75%                      3.500000                0.000000   
max                 311531.400000             2357.000000   

       Трафик голосовой, дни  Трафик передачи данных, Гб  
count            2457.000000                 2457.000000  
mean                0.070818                    0.195767  
std                 0.598477                    6.429233  
min                 0.000000                    0.000000  
25%                 0.000000                    0.000000  
50%                 0.000000                    0.000000  
75%                 0.000000                    0.000000  
max                 7.000000                  304.000000

<div style="border-radius: 15px; border: 3px solid PaleGreen; padding: 15px;">
<b> Сегмент 7</b>
  
    Категорией устройства "data_card" пользуется более 2000 абонентов. 
    При этом абоненты данного вида устройства в основном пользуются голосовым трафиком
</div>

In [136]:
(df_1
 .pivot_table(index= 'Категория устройства', 
              values =['Сумма начислений в месяц, руб', 'Трафик голосовой, мин', 'Трафик передачи данных, Мб', 'Трафик в роуминге, мин' ], 
              aggfunc = 'sum'
             ).sort_values(by='Сумма начислений в месяц, руб', ascending=False).reset_index()
)

Категория устройства  Сумма начислений в месяц, руб  Трафик в роуминге, мин  \
0           smartphone                     42024035.4               1250841.7   
1                phone                      2994610.4                102662.0   
2                modem                      1716530.8                   711.5   
3               router                       558239.6                  1742.1   
4            data card                       297153.3                  2427.5   
5               tablet                         6206.2                  3177.8   
6            usb modem                         2370.7                     0.0   
7       network device                          263.2                     0.7   

   Трафик голосовой, мин  Трафик передачи данных, Мб  
0             11289525.5                 307072264.0  
1              2026703.4                    244228.2  
2                71751.5                  28030301.0  
3                30675.3                   4948215.5  
4               256174.8                    507699.3  
5                27139.7                      4323.4  
6                 6402.6                         0.0  
7                39428.0                         0.0

<div style="border-radius: 15px; border: 3px solid PaleGreen; padding: 15px;">
<b> Итог</b>
  
    1. Пользователи, которые пользуются категорией устройства 'phone' чаще всего совершают звонки в сети и в роуминге, реже   пользуются интернетом.
    2.  Портрет пользователя "smartphone" разнообразный, здесь вовлечены все основные показатели.
    Средние показатели составляют:
        + Трафик передачи данных, Гб - 36.7
        + Трафик голосовой, мин - 1382
        + Трафик в роуминге, мин - 153.1
    3. "Usb modem" в выборке не пользуется популярность, всего 1 абонент использует для звонков.
    При этом голосовой трафик достигает 6000 мин, что превышает среднее значение более чем в 4 раза 
    4. Пользователи "Router" преимущественно пользуются интернетом
    5. Категория устройства "modem" пользуется популярностью, более 3000 абонентов. 
    При этом абоненты данного вида устройства используют преимущественно трафик передачи данных
    6. Категорией устройства "tablet" пользуется всего 4 человека из выборки. 
    При этом абоненты данного вида устройства активно используют голосовой трафик и трафик в роуминге
    7. Категорией устройства "network_device" пользуется всего 3 человека из выборки. 
    При этом абоненты данного вида устройства используют преимущественно голосовой трафик
    8.Категорией устройства "data_card" пользуется более 2000 абонентов. 
    При этом абоненты данного вида устройства в основном пользуются голосовым трафиком
</div>

<div style="border-radius: 15px; border: 3px solid PaleGreen; padding: 15px;">
<b> Вывод</b>
  
    1. Из всего набора категорий устройств можно выделить "smartphone". Здесь наибольшее количество пользователей. 
    Для этой категории на основе среднего, медианного значения, а также перцинтелей можно формировать пакеты предложений для разной категории пользователей.
    Так к примеру базовый тариф брать как 25%, основной на основе медианного значения (на него сделать акцент при формировании рекламных сценариев)
    2. В отдельную группу выделить пользователей "router" и "modem". Две категории в которых абоненты пользуются в основном 
    интернет трафиком, медианные значения начислений примерно одинакы и равны 507 и 489 соответсвенно, при этом число пользователей
    сильно разные и соотносятся примерно 1:6, что говорит о возможности расширения объемов продукта "router"
    3. Категории "tablet", "usb modem", "network device" приносят мало денег и пользуются небольшой популярностью, тут возможно
    стоит отказаться от продукта и предлодить пользователям перейти на альтернативный продукт
</div>